src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -2725,7 +2722,7 @@ public J visitProperty(KtProperty property, ExecutionContext data) {
             maybeBeforeSemicolon = prefix(property.getLastChild());
         }
 
-        JavaType.Variable vt = variableType(property);
+        JavaType.Variable vt = variableType(property, owner(property));
```

Comment:
Since we now only ever use `owner()` like here (either `variableType(psi, owner(psi))` or `type(psi, owner(psi))`), we could also consider doing away with the `ownerStack` and instead just search the PSI tree upwards. Then we would also not need a separate parameter for the owner, but could just do that inside the `PsiElementAssociations` class. There is of course a slight performance overhead with this (always searching the tree upwards), but I think it should be pretty negligible.

I don't have any strong feelings about this. I just wanted to point it out.

Reply:
I think that's a great point, it should be pretty simple for us to find the parent `KtDeclaration` instead of tracking the owner. For now, I prefer to leave that as a separate change and focus on TypeMapping optimizations.